utils-core/main/default/lwc/datatable/datatable.js

```diff
@@ -92,6 +93,14 @@ export default class Datatable extends LightningElement {
         this._sortableFields = createFlattenedSetFromDelimitedString(value, ',');
     }
 
+    @api
+    get columnLabels() {
+        return this._columnLabels;
+    }
+    set columnLabels(value = '') {
+        this._columnLabels = value === '' ? [] : value.split(',');
```

Comment:
I think we should frontload the setter processing and construct a private JS Map so that when we're constructing the columns themselves, all it's doing is using a `_columnLabelMap.has(col.fieldApiName)` then going and doing the replacement.

In other words, `_setFieldLabel` logic can go in this setter.

My opinion here is that the setter itself could do more processing, it's more readable for a future developer to not have to go down to line 5xx or 6xx to understand "ah, there is some magic here".

Hiding magic is not always good ;)

Some psuedo, i have no clue if this compiles or works but i wrote it real quick to show you how I'd like to see this processed:

```
set columnLabels(value = '') {
    if (value !== '') {
        const columnMappingArr = value.trim().split(',');
        // When this has keys, it will be used to reassign labels during column construction
        this._columnLabelsMap = new Map(
            columnMappingArr
                .filter(mapping => mapping.includes(COLUMN_LABEL_DELIMITER))
                .map(mapping => {
                    // Gets rid of any excess spaces in and around the delimiter
                    const mapParts = mapping.split(COLUMN_LABEL_DELIMITER).forEach(part => part.trim());
                    // Shorthand to instantiate map from array
                    return [mapParts[0], mapParts[1]];
                });
        );
    } else {
        this._columnLabelsMap = new Map();
    }
}
```

Reply:
That sounds totally reasonable to me — it shall be done. Thank you, as well, for writing that out! The rationale helps as much as the pseudo-code